lib/connectors/memory.js

```diff
@@ -461,6 +460,15 @@ function applyFilter(filter) {
     if (typeof value === 'string' && (example instanceof RegExp)) {
       return value.match(example);
     }
+
+    if (example.regexp) {
+      var normalizedRegExp = utils.toRegExp(example.regexp);
+      if (normalizedRegExp instanceof Error)
```

Comment:
Makes sense. I will dig into the root cause and make sure dao provides the correct data type to the connector.